examples/proc_GDS_pulse_capture.py

```diff
@@ -12,147 +12,160 @@
 import numpy as np
 from itertools import cycle
 
+colorcyc_list = plt.rcParams["axes.prop_cycle"].by_key()["color"]
 color_cycle = cycle(
-    [
-        "#1f77b4",
-        "#ff7f0e",
-        "#2ca02c",
-        "#d62728",
-        "#9467bd",
-        "#8c564b",
-        "#e377c2",
-        "#7f7f7f",
-        "#bcbd22",
-        "#17becf",
-    ]
-)
+    colorcyc_list
+)  # this can be done more than once to spin up multiple lists
 
 atten_ratio = 101.35  # attenutation ratio
-show_all = False  # diagnostic
+skip_plots = 10  # diagnostic -- set this to None, and there will be no plots
 with psd.figlist_var() as fl:
     for filename, nodename, amplitude in [
         ("240730_test_amp1_fin_pulse_calib.h5", "pulse_calib_1", 1.0),
         ("240730_test_amp0p1_fin_pulse_calib.h5", "pulse_calib_1", 0.1),
     ]:
+        fl.basename = f"amplitude = {amplitude}"
         d = psd.find_file(
             filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
         )
-        label = "amplitude = %f" % amplitude
-        thiscolor = next(color_cycle)
-        # {{{ allocate shapes for the final t_v_beta and beta_v_t plots
-        t_v_beta = d.shape.pop("t").alloc().rename("p_90", "beta")
-        t_v_beta.setaxis("beta", d.get_prop("desired_betas"))
-        beta_v_t = d.shape.pop("t").alloc().rename("p_90", "At_p")
-        beta_v_t.setaxis(
-            "At_p",
-            list(
-                d.get_prop("acq_params")["amplitude"] * d.get_prop("set_p90s")[j]
-                for j in range(len(d["p_90"]))
-            ),
-        )
+        d.set_units("t", "s")  # why isn't this done already??
+        d.rename(
+            "p_90", "t_pulse"
+        )  # just b/c existing name doesn't make sense
+        d["t_pulse"] = np.float64(
+            d["t_pulse"]
+        )  # PR COMMENT: these seem to be integers.  Since the pulse lengths should have been created with linspace, that is super super suspect
```

Comment:
see and remove PR comments after dealing with them.

The data needs to be reacquired b/c:

- why is this axis labeled with integers? Very suspicious.
- The raw data has a very stepped characteristic -- it looks like you are zoomed way out on the y-axis.
- I would just change the name of the indirect dimension to t_pulse so we don't have to do this.
- Is there an open PR for saving the units to the file?

Reply:
PR [here](https://github.com/jmfrancklab/pyspecdata/pull/50)